sentry/src/main/java/io/sentry/SentryItemType.java

```diff
@@ -12,6 +12,18 @@
 
   private final String itemType;
 
+  public static SentryItemType resolve(Object obj) {
+    if (obj instanceof SentryEvent) {
+      return Event;
+    } else if (obj instanceof Transaction) {
+      return Transaction;
+    } else if (obj instanceof Session) {
+      return Session;
+    } else {
+      return Attachment;
+    }
```

Comment:
I'd test `instanceof` here too and the `else` would be returning `null` just for the sake of safety, a `null` obj would return `Attachment` right now

Reply:
It would be instance of which class? My understanding is that anything can be attached as an attachment.

Reply:
```
} else if (obj instanceof Attachment) {
      return Attachment;
} else {
      return null;
}
```

and who calls `SentryItemType resolve(Object obj)` and returns null, should actually discard the type, not sure if its a good idea that all non Event, Transaction and Session become an Attachment, eg UserFeedback will soon be out, it doesnt make sense to me to be an Attachment, unless this is a default fallback that I'm not aware of, its not a bad idea, just thinking this could lead to other issues

Reply:
I think it has to wait. There is no `Attachment` class at the moment..?

Reply:
no, we don't support Attachment yet, good point, in that case, we could even remove the SentryItemType.Attachment altogether.

Reply:
Considering there are already tests that use attachments and it's been there independently of my changes, I won't touch it at the moment.